src/command/process/RawToProfProcessCommand.java

```diff
@@ -0,0 +1,181 @@
+            Command.validateName(file.getInfile(), MaxLength.FILE_FILENAME, "Infile");
+            Command.validateName(file.getOutfile(), MaxLength.FILE_FILENAME, "Outfile");
+            Command.validateName(file.getGenomeVersion(), MaxLength.GENOME_VERSION, "Genome version");
+        }
+    }
+
+    /**
+     * Run through the list of RawToProfProcessFiles and run processing on each with filePath as parameter.
+     * @param filePath associated with expId
+     */
+    @Override
+    public void doProcess(Map.Entry<String,String> filePath) {
+        for(RawToProfProcessFile file: files) {
+            file.ProcessFile(filePath);
+        }
+    }
+
+    public ArrayList<RawToProfProcessFile> getFiles() {
+        return files;
+    }
+
+    @Override
+    public String toString() {
+        return "RawToProfileCommand{" +
+                "files=" + files +
+                '}';
+    }
+    @Expose
+    private ArrayList<RawToProfProcessFile> files;
+
+    public class RawToProfProcessFile {
+
+        /**
+         * Class is used to start a single raw to profile processing with correct parameters.
+         */
+
+        @Expose
+        private String infile;
+
+        @Expose
+        private String outfile;
+
+        @Expose
+        private String params;
+
+        @Expose
+        private String genomeVersion;
+
+        @Expose
+        private boolean keepSam;
+
+
+        public String getInfile() {
+            return infile;
+        }
+
+        public String getOutfile() {
+            return outfile;
+        }
+
+        public String getParams() {
+            return params;
+        }
+
+        public String getGenomeVersion() {
+            return genomeVersion;
+        }
+
+        public boolean shouldKeepSam() {
+            return keepSam;
+        }
+
+        @Override
+        public String toString() {
+            return "RawToProfProcessFile{" +
+                   "infile='" + infile + '\'' +
+                   ", outfile='" + outfile + '\'' +
+                   ", params='" + params + '\'' +
+                   ", genomeVersion='" + genomeVersion + '\'' +
+                   ", keepSam=" + keepSam +
+                   '}';
+        }
+
+        /**
+         * Call upon a single raw to profile processing with correct parameters.
+         * @param filePaths
+         */
+        public void ProcessFile(Map.Entry<String,String> filePaths) {
+            DatabaseAccessor db = null;
+
+            try {
+                db = initDB();
+
+                //Get the genome information from the database.
+                Genome g = db.getGenomeRelease(getGenomeVersion());
+
+                if (g == null) {
+                    throw new UnsupportedOperationException("Could not find genome version: " +
+                            getGenomeVersion());
+                }
+                else {
+                    //Get the path of the genome.
+                    String genomeFolderPath = g.folderPath;
```

Comment:
Accessing Genome field variable folderPath directly. Although final, this isn't pretty.